lib/smack/SmackWarnings.cpp

```diff
@@ -0,0 +1,34 @@
+#include "llvm/Support/Debug.h"
+#include "smack/SmackWarnings.h"
+
+
+namespace smack {
+using namespace llvm;
+void SmackWarnings::warnWithType(const char* type, std::string msg) {
+  if (isCurrentDebugType(type))
```

Comment:
Why do you need this if condition? Could you not use the LLVM macro instead?

Reply:
It appears that the development process suggests what @keram88 proposed as opposed to using LLVM debug system. I think macros like `DEBUG_WITH_TYPE` creates names (i.e., the first arguments) that may conflict with other names, which may not be checked at compile time. W.r.t this condition, it's purpose is to bypass the message generation statements if the current warning level is too low.